gobblin-runtime/src/main/java/gobblin/runtime/api/JobSpecMonitorFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.runtime.api;
+
+/**
+ * A fact
+ */
+public interface JobSpecMonitorFactory {
+  /**
+   * Add a {@link JobSpecMonitor} that will be notified of new jobs. {@link JobSpecMonitor}
+   * will call put for any new / updated jobs, and remove for any deleted jobs.
+   * @param instanceDriver    the GobblinInstanceDriver managing the job catalog; this can be
+   *                          used to access the environment for configuration
+   * @param  jobCatalog       the job catalog to be notified for JobSpec operations.
+   */
+  JobSpecMonitor monitorJobCatalog(GobblinInstanceDriver instanceDriver, MutableJobCatalog jobCatalog);
```

Comment:
I think the name is reversed. The `JobMonitor` does not monitor the `JobCatalog`. Maybe call the method `forJobCatalog`?